src/components/BottomMenuModal/SendModal/SendModalTokensTabView.tsx

```diff
@@ -103,6 +127,218 @@ const SendModalTokensTabView = ({ payload }: { payload?: SendModalData }) => {
+      );
+      const costAsFiat = +estimatedCost * nativeAssetPrice;
+      const feeTokenPrice = await getPrices([selectedFeeAsset.token]);
+      let estimatedCostInToken;
+      if (feeTokenPrice) {
+        const feeTokenPriceInUSD = feeTokenPrice[0].usd;
+        estimatedCostInToken = (costAsFiat / feeTokenPriceInUSD).toFixed(
+          selectedFeeAsset.decimals
+        );
+        setFeeMin(estimatedCostInToken);
+        setApproveData(
+          encodeFunctionData({
+            abi: erc20Abi,
+            functionName: 'approve',
+            args: [
+              selectedPaymasterAddress as Address,
+              parseUnits(estimatedCostInToken, selectedFeeAsset.decimals),
+            ],
+          })
+        );
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (!selectedAsset) return;
+    if (!gasPrice) return;
+    let gasCost = 0;
+    /*
+     * The gas cost is estimated based on the type of asset being sent.
+     * The gas cost is calculated based on the type of asset being sent
+     * and the chain ID of the asset. The gas cost is then used to calculate
+     * the estimated cost in fiat.
+     * And the token cost is generally estimated as $0.01 even for undeployed wallet
+     * though it can be even lower for deployed wallet to save rpc call for checking
+     * deployed wallet or not, we can use the same gas cost for both deployed and undeployed wallet
+     */
+    if (selectedAsset.chainId === 42161) {
+      if (selectedAsset.type === 'token') {
+        if (selectedAsset.asset.contract === ethers.constants.AddressZero) {
+          gasCost = Number(
+            process.env.REACT_APP_NATIVE_GAS_CONSUMPTION_ARBITRUM
+          ); // estimated gas consumption for native asset transfer for undeployed wallet + 15% markup
+        } else {
+          gasCost = Number(
+            process.env.REACT_APP_TOKEN_GAS_CONSUMPTION_ARBITRUM
+          ); // estimated gas consumption for token asset transfer for undeployed wallet + 15% markup
+        }
+      } else if (selectedAsset.type === 'nft') {
+        gasCost = Number(process.env.REACT_APP_NFT_GAS_CONSUMPTION_ARBITRUM); // estimated gas consumption for token asset transfer for undeployed wallet + 15% markup
+      }
+    } else {
+      // eslint-disable-next-line no-lonely-if
+      if (selectedAsset.type === 'token') {
+        if (selectedAsset.asset.contract === ethers.constants.AddressZero) {
+          gasCost = Number(process.env.REACT_APP_NATIVE_GAS_CONSUMPTION); // estimated gas consumption for native asset transfer for deployed wallet + 15% markup
+        } else {
+          gasCost = Number(process.env.REACT_APP_TOKEN_GAS_CONSUMPTION); // estimated gas consumption for token asset transfer for deployed wallet + 15% markup
+        }
+      } else if (selectedAsset.type === 'nft') {
+        gasCost = Number(process.env.REACT_APP_NFT_GAS_CONSUMPTION); // estimated gas consumption for token asset transfer for deployed wallet + 15% markup
```

Comment:
The same applies for these process.env values

Reply:
The same ans from above works here too